src/main/java/se/liu/ida/hefquin/engine/datastructures/impl/SolutionMappingsHashTable.java

```diff
@@ -1,85 +1,155 @@
 package se.liu.ida.hefquin.engine.datastructures.impl;
 
-import java.util.Iterator;
+import java.util.*;
 
 import org.apache.jena.graph.Node;
 import org.apache.jena.sparql.core.Var;
 
+import org.apache.jena.sparql.engine.binding.Binding;
 import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.queryplan.ExpectedVariables;
 
 public class SolutionMappingsHashTable extends SolutionMappingsIndexBase
 {
+	protected final Map<List<Node>, List<SolutionMapping>> map = new HashMap<>();
+	protected final Set<Var> joinVariables;
+
+	public SolutionMappingsHashTable(final ExpectedVariables... inputVars){
```

Comment:
What's the purpose of passing potentially multiple `ExpectedVariables` objects to this constructor? Isn't it enough to pass a `Set<Var>` instead?

Reply:
Updated. Pass a Set<Var> to this constructor.